infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -157,6 +159,63 @@ public String getNodeIdBase64() {
         return (T) getServerConfig().get(optionName);
     }
 
+
+    /** Install a single node layout if and only if no layout is currently installed.
+     *  Synchronized, so this method is thread-safe.
+     *
+     *  @return True, if a new layout was installed, false otherwise.
+     */
+    public synchronized boolean installSingleNodeLayoutIfAbsent() {
+        if ((Boolean) getServerConfig().get("--single") && getCurrentLayout() == null) {
+            setCurrentLayout(getNewSingleNodeLayout());
+            return true;
+        }
+        return false;
+    }
+
+    /** Get a new single node layout used for self-bootstrapping a server started with
+     *  the -s flag.
+     *
+     *  @returns A new single node layout with a unique cluster Id
+     *  @throws IllegalArgumentException    If the cluster id was not auto, base64 or a UUID string
+     */
+    public Layout getNewSingleNodeLayout() {
+        final String clusterIdString = (String) getServerConfig().get("--cluster-id");
+        UUID clusterId;
+        if (clusterIdString.equals("auto")) {
+            clusterId = UUID.randomUUID();
+        } else {
+            // Is it a UUID?
+            try {
+                clusterId = UUID.fromString(clusterIdString);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "clusterId". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)